tests/security/selinux/sestatus.pm

```diff
@@ -17,29 +17,26 @@
 # Maintainer: llzhao <llzhao@suse.com>
 # Tags: poo#40358, tc#1682592
 
-use base "consoletest";
+use base 'opensusebasetest';
 use power_action_utils "power_action";
+use bootloader_setup 'add_grub_cmdline_settings';
 use strict;
 use testapi;
 use utils;
 
 sub run {
     my ($self) = @_;
-    select_console "root-console";
+    $self->select_serial_terminal;
 
     # SELinux by default
     validate_script_output("sestatus", sub { m/SELinux status: .*disabled/ });
 
-    # SELinux enabled
-    assert_script_run("sed -i '/^GRUB_CMDLINE_LINUX_DEFAULT/s/\"/& security=selinux selinux=1 enforcing=0 /' /etc/default/grub");
-    assert_script_run("grub2-mkconfig -o /boot/grub2/grub.cfg");
+    # enable SELinux in grub
+    add_grub_cmdline_settings('security=selinux selinux=1 enforcing=0', 1);
```

Comment:
it is a first and only usage of this function , and because only thing this function is doing is bypassing arguments to `change_grub_config` function can you explain why we need it ?

Reply:
I thought it's clearer approach than everyone using `sed` and calling `grub2-mkconfig` directly.

Reply:
```suggestion
    change_grub_config('"$',' security=selinux selinux=1 enforcing=0"', get_cmdline_var(), "g",  1);
```

yeah sure I have no problems with your changes to `change_grub_config` . But what you will loose if you just do this here and `add_grub_cmdline_settings` delete function at all ?

Reply:
Increase readability: in `add_grub_cmdline_settings('security=selinux selinux=1 enforcing=0', 1);` is obvious from name, that I'm adding something to grub and just have to see what means `1`.

Parameters in `change_grub_config('"$',' security=selinux selinux=1 enforcing=0"', get_cmdline_var(), "g",  1);` aren't really readable, you have to look into implementation, what `'"$','`, `"g"` and `get_cmdline_var()` mean. Having only `change_grub_config()` one would say "lets use `sed` in `assert_script_run` instead" as it was before. I meant `change_grub_config()` for more complicated manipulation with grub config, but why bother user with the details?

Reply:
I agree with @pevik. We should always make the function / API clearly and simple. And it is better to focus on a specific usage and reflected on the name. We should not expect the developer do more 'regex works'.

Reply:
ok